src/random_test.cpp

```diff
@@ -2679,123 +2679,213 @@ void Table::Compare_between_engine(const std::string &sql, Thd1 *thd) {
   set_default();
 }
 
+
 bool execute_sql(const std::string &sql, Thd1 *thd) {
-  auto query = sql.c_str();
-  static auto log_all = opt_bool(LOG_ALL_QUERIES);
-  static auto log_failed = opt_bool(LOG_FAILED_QUERIES);
-  static auto log_success = opt_bool(LOG_SUCCEDED_QUERIES);
-  static auto log_query_duration = opt_bool(LOG_QUERY_DURATION);
-  static auto log_client_output = opt_bool(LOG_CLIENT_OUTPUT);
-  static auto log_query_numbers = opt_bool(LOG_QUERY_NUMBERS);
-  std::chrono::system_clock::time_point begin, end;
+    auto query = sql.c_str();
+    static auto log_all = opt_bool(LOG_ALL_QUERIES);
+    static auto log_failed = opt_bool(LOG_FAILED_QUERIES);
+    static auto log_success = opt_bool(LOG_SUCCEDED_QUERIES);
+    static auto log_query_duration = opt_bool(LOG_QUERY_DURATION);
+    static auto log_client_output = opt_bool(LOG_CLIENT_OUTPUT);
+    static auto log_query_numbers = opt_bool(LOG_QUERY_NUMBERS);
+    std::chrono::system_clock::time_point begin, end;
+
+    if (log_query_duration) {
+        begin = std::chrono::system_clock::now();
+    }
 
-  if (log_query_duration) {
-    begin = std::chrono::system_clock::now();
-  }
+#ifdef DUCKDB
+    try {
+        // Create DuckDB connection
+        duckdb::DuckDB db(nullptr);  // In-memory database
```

Comment:
For each SQL to execute , It create new in memory database. This is not what is required here. 
 We have to create  in memory database just once. use connection based on --thread we have to open a connection against that.